gating/pre_merge_test/run_system_tests.sh

```diff
@@ -19,30 +21,65 @@ SYS_TEST_SOURCE_REPO="${SYS_TEST_SOURCE_BASE}/${SYS_TEST_SOURCE}"
 SYS_TEST_BRANCH="${SYS_TEST_BRANCH:-master}"
 export SYS_INVENTORY="/opt/openstack-ansible/playbooks/inventory"
 
+## Functions -----------------------------------------------------------------
+
+# Update the RC return code value unless it has previously been set to a
+# non-zero value.
+update_return_code() {
+    if [ "$RC" -eq "0" ]; then
+        RC=$1
+    fi
+}
+
+# Return the RC return code value unless it has not previously been set.
+# If that is the case, pass through the exit code of the last call.
+my_exit() {
+    if [ "$RC" -eq "0" ]; then
+        exit $1
+    else
+        exit $RC
+    fi
+}
+
 ## Main ----------------------------------------------------------------------
 
+# Trap script termination to return a captured RC value without prematurely
+# terminating the script.
+trap 'my_exit $?' INT TERM EXIT
```

Comment:
why does INT need to be trapped?

Reply:
In fact, since nothing is actually being done (and the my_exit function is only run once (and even then isn't really needed)) this trap isn't needed.